src/Symfony/Component/Runtime/Internal/SymfonyErrorHandler.php

```diff
@@ -24,12 +24,31 @@ class SymfonyErrorHandler
 {
     public static function register(bool $debug): void
     {
-        BasicErrorHandler::register($debug);
+        if (!class_exists(ErrorHandler::class)) {
+            BasicErrorHandler::register($debug);
 
-        if (class_exists(ErrorHandler::class)) {
+            return;
+        }
+
+        error_reporting(-1);
+
+        if (!\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true)) {
+            ini_set('display_errors', $debug);
+        } elseif (!filter_var(\ini_get('log_errors'), \FILTER_VALIDATE_BOOL) || \ini_get('error_log')) {
+            // CLI - display errors only if they're not already logged to STDERR
+            ini_set('display_errors', 1);
+        }
+
+        if (0 <= \ini_get('zend.assertions')) {
+            ini_set('zend.assertions', 1);
+            ini_set('assert.active', $debug);
+            ini_set('assert.exception', 1);
+        }
```

Comment:
inlining what `BasicErrorHandler::register() does currently (prevents the call to `restore_error_handler();`, removed below)

Reply:
should we create an internal public method in the BasicErrorHandler that we could reuse instead of duplicating the logic ?

Reply:
I wondered about this and thought this wasn't necessary: maintaining these duplicated lines shouldn't be a problem. Inlining prevents loading one more class on the hotpath.